### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by aprocter**
Intel folks: I'll be opening up an internal thread to discuss this, including more on implications and motivation. Please keep detailed discussions about this there.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# that applies (if I am reading the docs correctly).
+
+# Default rule: @diyessi gets everything.
+*					@diyessi
+
+/cmake/					@rkimballn1 @silee2
+
+/.ci/					@crlishka
+/contrib/docker/			@crlishka
+/.travis.yml				@crlishka @postrational
```
Comment (crlishka) [inline CODEOWNERS]: Let's give /.travis.yml to @postrational only.  I know almost nothing about how Travis CI is running, as it was developed by Michal.

Comment (aprocter) [reply CODEOWNERS]: Done

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Remember that the last applicable rule for any given file is the only one
+# that applies (if I am reading the docs correctly).
+
+# Default rule: @diyessi gets everything.
+*                                   @diyessi
+
+/cmake/                             @rkimballn1 @silee2
+
+/.ci/                               @crlishka
```
Comment (postrational) [inline CODEOWNERS]: We may need to split `.ci` into three parts. I can be responsible for `travis` and `onnx` subdirs

Comment (aprocter) [reply CODEOWNERS]: @postrational @crlishka What do you guys think about:

```
/.ci/                               @crlishka
/.ci/travis                         @postrational
/.ci/onnx                           @postrational
```

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+/contrib/docker/                    @crlishka
+/.travis.yml                        @postrational
+
+/.clang-format                      @rkimballn1
+/.gitattributes                     @rkimballn1
+/.gitignore                         @rkimballn1
+/VERSION.in                         @rkimballn1 @silee2
+
+/maint/                             @rkimballn1
```
Comment (rkimballn1) [inline CODEOWNERS]: Not me. I had nothing to do with this

File: CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+/src/resource/                      @rkimballn1
+/src/tools/                         @rkimballn1
+/src/ngraph/autodiff/               @diyessi
+/src/ngraph/builder/                @mbrookhart
+/src/ngraph/codegen/                @rkimballn1
+/src/ngraph/frontend/               @postrational
+/src/ngraph/op/                     @diyessi @aprocter
+/src/ngraph/pass/                   @Krovatkin
+/src/ngraph/pattern/                @Krovatkin
+/src/ngraph/runtime/                @rkimballn1 @jbobba
```
Comment (rkimballn1) [inline CODEOWNERS]: I think this should be just me, just the files in this dir, not subdirs

File: CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+/src/ngraph/runtime/interpreter/    @rkimballn1
+/src/ngraph/runtime/reference/      @aprocter
+/src/ngraph/type/                   @diyessi
+/src/ngraph/serializer.*pp          @rkimballn1
+
+# Putting this last so it's not overridden by directory rules
+CMakeLists.txt                      @rkimballn1 @silee2
+
+# Putting this last to make sure nothing else overrides.
+/CODEOWNERS                         @diyessi
```
Comment (rkimballn1) [inline CODEOWNERS]: Why not Adam Proctor for this ?

File: CODEOWNERS (hunk 3 above)
Comment (avijit-nervana) [reply CODEOWNERS]: I think /maint is @crlishka

File: CODEOWNERS (hunk 5 above)
Comment (aprocter) [reply CODEOWNERS]: Talked to @diyessi and we're thinking it makes the most sense to have him on it. He can always tag me if there's some issue about the formatting of the file or something.

**Comment by aprocter** [comment]
NB: turns out I was _wrong_ about admin powers being required for the auto-tagging. As soon as we merge, owners will automatically be tagged on new PRs for stuff they own.

**Comment by aprocter** [comment]
> We need owners for 'test' or figure out a way to split it across different components.

Yeah, we should think about how to split this up. We can certainly mark individual files within that directory, or we could possibly start to split the test files up into subdirectories. (I've thought for a while that the latter might be nice... it's getting pretty crowded in there.)

**Comment by aprocter** [comment]
@cconvey Gotcha. That makes sense. I think I will just delete that line so it defaults to @diyessi.

**Pull request merged**